### PR TITLE
feat: add `supabase unlink` command

### PR DESCRIPTION
**Pull request by eleijonmarck**
NOTE:
- [ ] tests are not ready yet, first I want confirmation that I am heading in the right direction.

## What kind of change does this PR introduce?
This introduces a new `unlink` cmd that unlinks the project set.

## What is the current behavior?
We have no way of "unlinking" a project today. This PR makes it so that we clean the .temp file from project-refs.

https://github.com/orgs/supabase/discussions/17991

## What is the new behavior?

```sh
(2) ~/dev/supabase/cli on main
% go run . link
Selected project ref: <project-ref>
Enter your database password (or leave blank to skip):
Finished supabase link.
(base)
(2) ~/dev/supabase/cli on main took 5s
% go run . unlink
Finished supabase unlink.
```
cleaned up the .temp files from project-refs with empty string.
```sh
(2) ~/dev/supabase/cli/supabase on main
% cd .temp
(base)
(2) ~/dev/supabase/cli/supabase/.temp on main
% ls
gotrue-version  pooler-url  postgres-version  project-ref  rest-version  storage-version
(base)
(2) ~/dev/supabase/cli/supabase/.temp on main
% cat rest-version
(base)
(2) ~/dev/supabase/cli/supabase/.temp on main
% cat project-ref
(base)
```

## Additional context

Add any other context or screenshots.

### CODE REVIEW

**Comment by sweatybridge** [comment]
Thank you for the contribution. This would be a much appreciated feature.

In addition to cleaning the `.temp` files, we should also purge any saved credentials from the platform native credentials store.

File: internal/unlink/unlink.go [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+	fmt.Fprintln(os.Stderr, "Local config differs from linked project. Try updating", utils.Bold(utils.ConfigPath))
+	enc := toml.NewEncoder(stdout)
+	enc.Indent = ""
+	if err := enc.Encode(updatedConfig); err != nil {
+		return errors.Errorf("failed to marshal toml config: %w", err)
+	}
+	return nil
+}
+
+func unlinkServices(ctx context.Context, fsys afero.Fs) {
```
Comment (sweatybridge) [inline internal/unlink/unlink.go:62]: Since unlink is a purely local operation, we can simply delete `utils.TempDir` without making any api requests.

Comment (eleijonmarck) [reply internal/unlink/unlink.go:62]: @sweatybridge i am working on a fix for this. The idea is to remove the `.temp` dir and input the `project-ref` to be able to remove the credentials for the project.

Comment (eleijonmarck) [reply internal/unlink/unlink.go]: @sweatybridge could you have a look at the [tests](https://github.com/supabase/cli/pull/1849/files#diff-f021fb6dc36f857212d1904d19a99399e15e565d4116eefa2971f66dca9f5c8fR13) i have written. they seem to fail due to:

```
        	Test:       	TestUnlinkCommand/unlink_valid_project
Removing credentials for project test-project...
    unlink_test.go:33: 
        	Error Trace:	/home/runner/work/cli/cli/internal/unlink/unlink_test.go:33
        	Error:      	Received unexpected error:
        	            	failed to remove credentials for project 'test-project': failed to set credentials: The name org.freedesktop.secrets was not provided by any .service files
```

Comment (sweatybridge) [reply internal/unlink/unlink.go]: I think you need to call `keyring.MockInit()` in this test suite. For eg.

https://github.com/supabase/cli/blob/main/internal/logout/logout_test.go#L19

**Comment by eleijonmarck** [comment]
@sweatybridge update the tests to confirm and i think this works for me when I unlink using supabase now. Could you test out to `unlink` a project

**Review by sweatybridge — APPROVED** [review]
I made some changes to handle missing keys in WSL environments. Thanks for the contribution!

**Pull request merged**